src/implementation.cpp

```diff
@@ -58,8 +58,6 @@ class detect_best_supported_implementation_on_first_use final : public implement
 
 const detect_best_supported_implementation_on_first_use detect_best_supported_implementation_on_first_use_singleton;
 
-internal::atomic_ptr<const implementation> active_implementation{&internal::detect_best_supported_implementation_on_first_use_singleton};
```

Comment:
There were two variables with that name: `simdjson::active_implementation` and `simdjson::internal::active_implementation`. That made the function `set_best()` choose implementation multiple times.